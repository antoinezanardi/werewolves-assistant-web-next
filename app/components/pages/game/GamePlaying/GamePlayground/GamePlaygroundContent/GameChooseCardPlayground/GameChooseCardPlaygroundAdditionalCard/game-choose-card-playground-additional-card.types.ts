import type { GameAdditionalCard } from "~/composables/api/game/types/game-additional-card/game-additional-card.class";

type GameChooseCardPlaygroundAdditionalCardProps = {
  additionalCard: GameAdditionalCard;
};

type GameChooseCardPlaygroundAdditionalCardEmits = {
  clickAdditionalCard: [GameAdditionalCard];
};

export type {
  GameChooseCardPlaygroundAdditionalCardProps,
  GameChooseCardPlaygroundAdditionalCardEmits,
};