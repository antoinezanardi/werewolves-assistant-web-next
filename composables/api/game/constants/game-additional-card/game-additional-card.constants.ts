import type { RoleName } from "~/composables/api/role/types/role.types";

const GAME_ADDITIONAL_CARDS_RECIPIENTS = ["thief", "actor"] as const satisfies Readonly<(RoleName)[]>;

export { GAME_ADDITIONAL_CARDS_RECIPIENTS };