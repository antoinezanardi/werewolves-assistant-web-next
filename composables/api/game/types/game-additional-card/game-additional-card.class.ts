import { Expose, plainToInstance } from "class-transformer";

import type { GameAdditionalCardRecipientRoleName } from "~/composables/api/game/types/game-additional-card/game-additional-card.types";
import type { RoleName } from "~/composables/api/role/types/role.types";
import { DEFAULT_PLAIN_TO_INSTANCE_OPTIONS } from "~/utils/constants/class-transformer.constants";

class GameAdditionalCard {
  @Expose()
  public _id: string;

  @Expose()
  public roleName: RoleName;

  @Expose()
  public recipient: GameAdditionalCardRecipientRoleName;

  @Expose()
  public isUsed: boolean;

  public static create(gameAdditionalCard: GameAdditionalCard): GameAdditionalCard {
    return plainToInstance(GameAdditionalCard, gameAdditionalCard, DEFAULT_PLAIN_TO_INSTANCE_OPTIONS);
  }
}

export { GameAdditionalCard };