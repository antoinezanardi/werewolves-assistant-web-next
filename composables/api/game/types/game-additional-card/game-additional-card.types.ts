import type { TupleToUnion } from "type-fest";

import type { GAME_ADDITIONAL_CARDS_RECIPIENTS } from "~/composables/api/game/constants/game-additional-card/game-additional-card.constants";

type GameAdditionalCardRecipientRoleName = TupleToUnion<typeof GAME_ADDITIONAL_CARDS_RECIPIENTS>;

export type { GameAdditionalCardRecipientRoleName };