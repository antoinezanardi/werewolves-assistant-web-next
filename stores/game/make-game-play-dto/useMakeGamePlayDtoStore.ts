import { defineStore } from "pinia";

import { MakeGamePlayTargetDto } from "~/composables/api/game/dto/make-game-play/make-game-play-target/make-game-play-target.dto";
import { MakeGamePlayVoteDto } from "~/composables/api/game/dto/make-game-play/make-game-play-vote/make-game-play-vote.dto";
import { MakeGamePlayDto } from "~/composables/api/game/dto/make-game-play/make-game-play.dto";
import type { RoleSides } from "~/composables/api/role/enums/role.enums";
import { StoreIds } from "~/stores/enums/store.enum";

const useMakeGamePlayDtoStore = defineStore(StoreIds.MAKE_GAME_PLAY_DTO, () => {
  const makeGamePlayDto = ref<MakeGamePlayDto>(MakeGamePlayDto.create({}));

  function setMakeGamePlayDto(makeGamePlayDtoValue: MakeGamePlayDto): void {
    makeGamePlayDto.value = MakeGamePlayDto.create(makeGamePlayDtoValue);
  }

  function resetMakeGamePlayDto(): void {
    makeGamePlayDto.value = MakeGamePlayDto.create({});
  }

  function addMakeGamePlayTargetDto(target: MakeGamePlayTargetDto): void {
    if (!makeGamePlayDto.value.targets) {
      makeGamePlayDto.value.targets = [];
    }
    makeGamePlayDto.value.targets.push(MakeGamePlayTargetDto.create(target));
  }

  function removeMakeGamePlayTargetDto(targetId: string): void {
    if (!makeGamePlayDto.value.targets) {
      return;
    }
    const targetIndex = makeGamePlayDto.value.targets.findIndex(target => target.playerId === targetId);
    if (targetIndex !== -1) {
      makeGamePlayDto.value.targets = makeGamePlayDto.value.targets.toSpliced(targetIndex, 1);
    }
    if (makeGamePlayDto.value.targets.length === 0) {
      makeGamePlayDto.value.targets = undefined;
    }
  }

  function addMakeGamePlayVoteDto(vote: MakeGamePlayVoteDto): void {
    if (!makeGamePlayDto.value.votes) {
      makeGamePlayDto.value.votes = [];
    }
    makeGamePlayDto.value.votes.push(MakeGamePlayVoteDto.create(vote));
  }

  function removeMakeGamePlayVoteDto(sourceId: string): void {
    if (!makeGamePlayDto.value.votes) {
      return;
    }
    const voteIndex = makeGamePlayDto.value.votes.findIndex(vote => vote.sourceId === sourceId);
    if (voteIndex !== -1) {
      makeGamePlayDto.value.votes = makeGamePlayDto.value.votes.toSpliced(voteIndex, 1);
    }
    if (makeGamePlayDto.value.votes.length === 0) {
      makeGamePlayDto.value.votes = undefined;
    }
  }

  function setDoesJudgeRequestAnotherVote(doesJudgeRequestAnotherVote: boolean | undefined): void {
    makeGamePlayDto.value.doesJudgeRequestAnotherVote = doesJudgeRequestAnotherVote;
  }

  function setChosenCardId(chosenCardId: string | undefined): void {
    makeGamePlayDto.value.chosenCardId = chosenCardId;
  }

  function setChosenSide(chosenSide: RoleSides | undefined): void {
    makeGamePlayDto.value.chosenSide = chosenSide;
  }
  return {
    makeGamePlayDto,
    setMakeGamePlayDto,
    resetMakeGamePlayDto,
    addMakeGamePlayTargetDto,
    removeMakeGamePlayTargetDto,
    addMakeGamePlayVoteDto,
    removeMakeGamePlayVoteDto,
    setDoesJudgeRequestAnotherVote,
    setChosenCardId,
    setChosenSide,
  };
});

export { useMakeGamePlayDtoStore };