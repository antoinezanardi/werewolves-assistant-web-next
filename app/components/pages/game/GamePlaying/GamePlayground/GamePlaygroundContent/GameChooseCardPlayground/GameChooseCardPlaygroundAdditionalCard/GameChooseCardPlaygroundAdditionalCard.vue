<template>
  <GlowElement
    id="game-choose-card-playground-additional-card"
  >
    <button
      id="additional-card-button"
      :aria-label="additionalCardButtonAriaLabel"
      :class="additionalCardButtonClasses"
      type="button"
      @click.prevent="onClickFromAdditionalCardButton"
    >
      <RoleImage
        id="additional-card-image"
        :class="additionalCardsRoleImageClasses"
        definition="normal"
        :role-name="additionalCard.roleName"
        :sizes="additionalCardImageSizes"
      />

      <span
        id="additional-card-role-label"
        class="flex flex-col font-semibold grow justify-center"
      >
        {{ additionalCardRoleNameLabel }}
      </span>
    </button>
  </GlowElement>
</template>

<script setup lang="ts">
import type { GameChooseCardPlaygroundAdditionalCardEmits, GameChooseCardPlaygroundAdditionalCardProps } from "~/components/pages/game/GamePlaying/GamePlayground/GamePlaygroundContent/GameChooseCardPlayground/GameChooseCardPlaygroundAdditionalCard/game-choose-card-playground-additional-card.types";
import RoleImage from "~/components/shared/role/RoleImage/RoleImage.vue";
import { useRoleName } from "~/composables/api/role/useRoleName";
import { useStrings } from "~/composables/misc/useStrings";
import { useAppBreakpoints } from "~/composables/style/useAppBreakpoints";
import { useMakeGamePlayDtoStore } from "~/stores/game/make-game-play-dto/useMakeGamePlayDtoStore";
import { useRolesStore } from "~/stores/role/useRolesStore";

const props = defineProps<GameChooseCardPlaygroundAdditionalCardProps>();

const emit = defineEmits<GameChooseCardPlaygroundAdditionalCardEmits>();

const { isSmallerThanMdBreakpoint } = useAppBreakpoints();

const additionalCardImageSizes = computed<string>(() => (isSmallerThanMdBreakpoint.value ? "75" : "125"));

const { makeGamePlayDto } = useMakeGamePlayDtoStore();

const rolesStore = useRolesStore();
const { getRoleSideForRoleName } = rolesStore;

const { getRoleNameLabel, getDefiniteRoleNameLabel } = useRoleName();

const { t } = useI18n();

const { lowerCaseFirstLetter } = useStrings();

const additionalCardRoleNameLabel = computed<string>(() => getRoleNameLabel(props.additionalCard.roleName));

const additionalCardButtonAriaLabel = computed<string>(() => {
  const lowerRecipientDefiniteRoleNameLabel = lowerCaseFirstLetter(getDefiniteRoleNameLabel(props.additionalCard.recipient, 1));

  return t("components.GameChooseCardPlaygroundAdditionalCard.chooseCardForRecipient", {
    roleName: additionalCardRoleNameLabel.value,
    recipientDefiniteName: lowerRecipientDefiniteRoleNameLabel,
  });
});

const additionalCardColor = computed<string>(() => {
  const roleSide = getRoleSideForRoleName(props.additionalCard.roleName);

  return roleSide === "villagers" ? "emerald" : "red";
});

const isAdditionalCardChosen = computed<boolean>(() => makeGamePlayDto.chosenCardId === props.additionalCard._id);

const additionalCardButtonClasses = computed<string>(() => {
  const baseClasses = `border-4 border-transparent flex flex-col glow:border-${additionalCardColor.value}-500 h-52 items-center me-2 p-3 rounded-lg`;
  if (isAdditionalCardChosen.value) {
    return `${baseClasses} !border-${additionalCardColor.value}-500`;
  }
  return baseClasses;
});

const additionalCardsRoleImageClasses = computed<string>(() => {
  const baseClasses = `glow:border-${additionalCardColor.value}-500 mb-1`;
  if (isAdditionalCardChosen.value) {
    return `${baseClasses} !border-${additionalCardColor.value}-500`;
  }
  return baseClasses;
});

function onClickFromAdditionalCardButton(): void {
  emit("clickAdditionalCard", props.additionalCard);
}
</script>