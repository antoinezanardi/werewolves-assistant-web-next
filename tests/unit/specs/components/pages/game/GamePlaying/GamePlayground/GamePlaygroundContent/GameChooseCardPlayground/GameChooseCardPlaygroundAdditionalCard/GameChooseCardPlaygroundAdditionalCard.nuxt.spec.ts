import { createTestingPinia } from "@pinia/testing";
import { createFakeCreateGameDto } from "@tests/unit/utils/factories/composables/api/game/dto/create-game/create-game.dto.factory";
import { createFakeMakeGamePlayDto } from "@tests/unit/utils/factories/composables/api/game/dto/make-game-play/make-game-play.dto.factory";
import { createFakeGameAdditionalCard } from "@tests/unit/utils/factories/composables/api/game/game-additional-card/game-additional-card.factory";
import type { mount } from "@vue/test-utils";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";
import type { GameChooseCardPlaygroundAdditionalCardProps } from "~/components/pages/game/GamePlaying/GamePlayground/GamePlaygroundContent/GameChooseCardPlayground/GameChooseCardPlaygroundAdditionalCard/game-choose-card-playground-additional-card.types";
import GameChooseCardPlaygroundAdditionalCard from "~/components/pages/game/GamePlaying/GamePlayground/GamePlaygroundContent/GameChooseCardPlayground/GameChooseCardPlaygroundAdditionalCard/GameChooseCardPlaygroundAdditionalCard.vue";
import type RoleImage from "~/components/shared/role/RoleImage/RoleImage.vue";
import { StoreIds } from "~/stores/enums/store.enum";
import { useMakeGamePlayDtoStore } from "~/stores/game/make-game-play-dto/useMakeGamePlayDtoStore";

const hoistedMocks = vi.hoisted(() => ({
  useRolesStore: {
    getRoleSideForRoleName: vi.fn(),
  },
}));

vi.mock("~/stores/role/useRolesStore.ts", () => ({
  useRolesStore: (): typeof hoistedMocks.useRolesStore => hoistedMocks.useRolesStore,
}));

describe("Game Choose Card Playground Additional Card Component", () => {
  let wrapper: ReturnType<typeof mount<typeof GameChooseCardPlaygroundAdditionalCard>>;
  const defaultCreateGameDto = createFakeCreateGameDto();
  const testingPinia = { initialState: { [StoreIds.CREATE_GAME_DTO]: { createGameDto: defaultCreateGameDto } } };

  const defaultProps: GameChooseCardPlaygroundAdditionalCardProps = {
    additionalCard: createFakeGameAdditionalCard({
      _id: "1",
      recipient: "thief",
      roleName: "seer",
      isUsed: false,
    }),
  };

  async function mountGameChooseCardPlaygroundAdditionalCardComponent(options: ComponentMountingOptions<typeof GameChooseCardPlaygroundAdditionalCard> = {}): Promise<ReturnType<typeof mount<typeof GameChooseCardPlaygroundAdditionalCard>>> {
    return mountSuspendedComponent(GameChooseCardPlaygroundAdditionalCard, {
      shallow: false,
      global: {
        plugins: [createTestingPinia(testingPinia)],
      },
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    hoistedMocks.useRolesStore.getRoleSideForRoleName.mockReturnValue("villagers");
    wrapper = await mountGameChooseCardPlaygroundAdditionalCardComponent();
    const makeGamePlayStore = useMakeGamePlayDtoStore();
    makeGamePlayStore.makeGamePlayDto = createFakeMakeGamePlayDto();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Additional Card Button", () => {
    it("should set base classes to button when card side is villagers.", () => {
      const button = wrapper.find<HTMLButtonElement>("#additional-card-button");
      const expectedClasses = [
        "border-4",
        "border-transparent",
        "flex",
        "flex-col",
        "glow:border-emerald-500",
        "h-52",
        "items-center",
        "me-2",
        "p-3",
        "rounded-lg",
      ];

      expect(button.classes()).toStrictEqual<string[]>(expectedClasses);
    });

    it("should set base classes to button when card side is werewolves.", async() => {
      hoistedMocks.useRolesStore.getRoleSideForRoleName.mockReturnValue("werewolves");
      wrapper = await mountGameChooseCardPlaygroundAdditionalCardComponent({
        props: {
          additionalCard: createFakeGameAdditionalCard({
            _id: "1",
            recipient: "thief",
            roleName: "werewolf",
            isUsed: false,
          }),
        },
      });
      const button = wrapper.find<HTMLButtonElement>("#additional-card-button");
      const expectedClasses = [
        "border-4",
        "border-transparent",
        "flex",
        "flex-col",
        "glow:border-red-500",
        "h-52",
        "items-center",
        "me-2",
        "p-3",
        "rounded-lg",
      ];

      expect(button.classes()).toStrictEqual<string[]>(expectedClasses);
    });

    it("should set base classes plus selected class to button when card side is villagers and selected.", async() => {
      wrapper = await mountGameChooseCardPlaygroundAdditionalCardComponent();
      const makeGamePlayStore = useMakeGamePlayDtoStore();
      makeGamePlayStore.makeGamePlayDto.chosenCardId = "1";
      await nextTick();
      const button = wrapper.find<HTMLButtonElement>("#additional-card-button");
      const expectedClasses = [
        "border-4",
        "border-transparent",
        "flex",
        "flex-col",
        "glow:border-emerald-500",
        "h-52",
        "items-center",
        "me-2",
        "p-3",
        "rounded-lg",
        "!border-emerald-500",
      ];

      expect(button.classes()).toStrictEqual<string[]>(expectedClasses);
    });

    it("should set base classes plus selected class to button when card side is werewolves and selected.", async() => {
      hoistedMocks.useRolesStore.getRoleSideForRoleName.mockReturnValue("werewolves");
      wrapper = await mountGameChooseCardPlaygroundAdditionalCardComponent({
        props: {
          additionalCard: createFakeGameAdditionalCard({
            _id: "1",
            recipient: "thief",
            roleName: "werewolf",
            isUsed: false,
          }),
        },
      });
      const makeGamePlayStore = useMakeGamePlayDtoStore();
      makeGamePlayStore.makeGamePlayDto.chosenCardId = "1";
      await nextTick();
      const button = wrapper.find<HTMLButtonElement>("#additional-card-button");
      const expectedClasses = [
        "border-4",
        "border-transparent",
        "flex",
        "flex-col",
        "glow:border-red-500",
        "h-52",
        "items-center",
        "me-2",
        "p-3",
        "rounded-lg",
        "!border-red-500",
      ];

      expect(button.classes()).toStrictEqual<string[]>(expectedClasses);
    });

    describe("Card Image", () => {
      it("should set card image to seer when card role name is seer.", () => {
        const roleImage = wrapper.findComponent<typeof RoleImage>("#additional-card-image");

        expect(roleImage.props("roleName")).toBe("seer");
      });

      it("should set base classes to card image when card side is villagers.", () => {
        const roleImage = wrapper.findComponent<typeof RoleImage>("#additional-card-image");
        const expectedClasses = [
          "border-4",
          "border-gray-800",
          "role-image",
          "rounded-lg",
          "h-[125px]",
          "w-[125px]",
          "glow:border-emerald-500",
          "mb-1",
        ];

        expect(roleImage.classes()).toStrictEqual<string[]>(expectedClasses);
      });

      it("should set base classes to card image when card side is werewolves.", async() => {
        hoistedMocks.useRolesStore.getRoleSideForRoleName.mockReturnValue("werewolves");
        wrapper = await mountGameChooseCardPlaygroundAdditionalCardComponent({
          props: {
            additionalCard: createFakeGameAdditionalCard({
              _id: "1",
              recipient: "thief",
              roleName: "werewolf",
              isUsed: false,
            }),
          },
        });
        const roleImage = wrapper.findComponent<typeof RoleImage>("#additional-card-image");
        const expectedClasses = [
          "border-4",
          "border-gray-800",
          "role-image",
          "rounded-lg",
          "h-[125px]",
          "w-[125px]",
          "glow:border-red-500",
          "mb-1",
        ];

        expect(roleImage.classes()).toStrictEqual<string[]>(expectedClasses);
      });

      it("should set base classes plus selected class to card image when card side is villagers and selected.", async() => {
        wrapper = await mountGameChooseCardPlaygroundAdditionalCardComponent();
        const makeGamePlayStore = useMakeGamePlayDtoStore();
        makeGamePlayStore.makeGamePlayDto.chosenCardId = "1";
        await nextTick();
        const roleImage = wrapper.findComponent<typeof RoleImage>("#additional-card-image");
        const expectedClasses = [
          "border-4",
          "border-gray-800",
          "role-image",
          "rounded-lg",
          "h-[125px]",
          "w-[125px]",
          "glow:border-emerald-500",
          "mb-1",
          "!border-emerald-500",
        ];

        expect(roleImage.classes()).toStrictEqual<string[]>(expectedClasses);
      });

      it("should set base classes plus selected class to card image when card side is werewolves and selected.", async() => {
        hoistedMocks.useRolesStore.getRoleSideForRoleName.mockReturnValue("werewolves");
        wrapper = await mountGameChooseCardPlaygroundAdditionalCardComponent({
          props: {
            additionalCard: createFakeGameAdditionalCard({
              _id: "1",
              recipient: "thief",
              roleName: "werewolf",
              isUsed: false,
            }),
          },
        });
        const makeGamePlayStore = useMakeGamePlayDtoStore();
        makeGamePlayStore.makeGamePlayDto.chosenCardId = "1";
        await nextTick();
        const roleImage = wrapper.findComponent<typeof RoleImage>("#additional-card-image");
        const expectedClasses = [
          "border-4",
          "border-gray-800",
          "role-image",
          "rounded-lg",
          "h-[125px]",
          "w-[125px]",
          "glow:border-red-500",
          "mb-1",
          "!border-red-500",
        ];

        expect(roleImage.classes()).toStrictEqual<string[]>(expectedClasses);
      });
    });

    describe("Card Label", () => {
      it("should translate card label to seer when card role name is seer.", () => {
        const label = wrapper.find<HTMLSpanElement>("#additional-card-role-label");

        expect(label.text()).toBe("shared.role.name.seer");
      });
    });

    describe("Emits", () => {
      it("should emit click additional card event when button is clicked.", async() => {
        const button = wrapper.find<HTMLButtonElement>("#additional-card-button");
        await button.trigger("click");

        expect(wrapper.emitted("clickAdditionalCard")).toBeTruthy();
      });
    });
  });
});