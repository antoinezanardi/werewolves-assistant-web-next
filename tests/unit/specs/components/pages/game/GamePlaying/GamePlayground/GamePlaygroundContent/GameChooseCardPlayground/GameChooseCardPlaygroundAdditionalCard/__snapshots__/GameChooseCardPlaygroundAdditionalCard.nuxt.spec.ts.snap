// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Choose Card Playground Additional Card Component > should match snapshot when rendered. 1`] = `
"<div class="grid glow" id="game-choose-card-playground-additional-card" style="--glow-top: 0px; --glow-left: 0px;">
  <div class="" style="grid-area: 1/1/1/1;"><button id="additional-card-button" aria-label="components.GameChooseCardPlaygroundAdditionalCard.chooseCardForRecipient, {" roleName":"shared.role.name.seer","recipientDefiniteName":"shared.role.definiteName.seer, 1"}" class="border-4 border-transparent flex flex-col glow:border-emerald-500 h-52 items-center me-2 p-3 rounded-lg" type="button"><img width="125" height="125" alt="shared.role.name.seer" data-nuxt-img="" class="border-4 border-gray-800 role-image rounded-lg h-[125px] w-[125px] glow:border-emerald-500 mb-1" id="additional-card-image" src="/svg/misc/infinite-spinner.svg"><span id="additional-card-role-label" class="flex flex-col font-semibold grow justify-center">shared.role.name.seer</span></button></div>
  <div class="glow-mask" glow="true" style="--glow-color: #f50057; grid-area: 1/1/1/1; pointer-events: none; mask: radial-gradient(var(--glow-size) var(--glow-size) at calc(var(--glow-x, -99999px) - var(--glow-left, 0px))
        calc(var(--glow-y, -99999px) - var(--glow-top, 0px)), #f50057 1%, transparent 50%);"><button id="additional-card-button" aria-label="components.GameChooseCardPlaygroundAdditionalCard.chooseCardForRecipient, {" roleName":"shared.role.name.seer","recipientDefiniteName":"shared.role.definiteName.seer, 1"}" class="border-4 border-transparent flex flex-col glow:border-emerald-500 h-52 items-center me-2 p-3 rounded-lg" type="button"><img width="125" height="125" alt="shared.role.name.seer" data-nuxt-img="" class="border-4 border-gray-800 role-image rounded-lg h-[125px] w-[125px] glow:border-emerald-500 mb-1" id="additional-card-image" src="/svg/misc/infinite-spinner.svg"><span id="additional-card-role-label" class="flex flex-col font-semibold grow justify-center">shared.role.name.seer</span></button></div>
</div>"
`;
